### ISSUE TITLE
Several CMake errors (with latest release on conda-forge)

### ISSUE THREAD

**Issue by tdegeus** [BlueBrain/HighFive#325]
* I'm getting a CMake error with the latest release (see below).
* Somehow the prefix `HIGFIVE_` has disappeared for xtensor (and I believe also for Eigen). E.g. `HIGFIVE_USE_XTENSOR` now has to be specified as `USE_XTENSOR`. I don't think this is wanted as it could lead to conflicts with other package. 

I think that once these issues are fixed a patch release should be made.

```cmake
set(HIGHFIVE_USE_INSTALL_DEPS 0)
set(HIGHFIVE_USE_BOOST 0)
# set(USE_XTENSOR 1)
find_package(HighFive REQUIRED)
```

gives

```none
CMake Warning (dev) at /Users/tdegeus/miniconda3/envs/covid_network/share/HighFive/CMake/HighFiveConfig.cmake:34 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'HIGHFIVE_USE_INSTALL_DEPS'.
Call Stack (most recent call first):
  CMakeLists.txt:16 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/tdegeus/miniconda3/envs/covid_network/share/HighFive/CMake/HighFiveConfig.cmake:46 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'HIGHFIVE_USE_BOOST'.
Call Stack (most recent call first):
  CMakeLists.txt:16 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

HIGHFIVE 2.2: (Re)Detecting Highfive dependencies (HIGHFIVE_USE_INSTALL_DEPS=NO)
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /Users/tdegeus/miniconda3/envs/covid_network/lib/libhdf5.dylib;/usr/lib/libpthread.dylib;/Users/tdegeus/miniconda3/envs/covid_network/lib/libz.dylib;/usr/lib/libdl.dylib;/usr/lib/libm.dylib (found version "1.10.5")
CMake Error at /Users/tdegeus/miniconda3/envs/covid_network/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR system serialization)
Call Stack (most recent call first):
  /Users/tdegeus/miniconda3/envs/covid_network/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /Users/tdegeus/miniconda3/envs/covid_network/share/cmake-3.17/Modules/FindBoost.cmake:2145 (find_package_handle_standard_args)
  /Users/tdegeus/miniconda3/envs/covid_network/share/HighFive/CMake/HighFiveTargetDeps.cmake:25 (find_package)
  /Users/tdegeus/miniconda3/envs/covid_network/share/HighFive/CMake/HighFiveConfig.cmake:63 (include)
  CMakeLists.txt:16 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/tdegeus/data/publications/covid/network_mean-field/analyses/width_depth/build/CMakeFiles/CMakeOutput.log".
```

**Comment by tdegeus** [comment]
Note:

```none
make --version
cmake version 3.17.0
```

**Comment by ferdonline** [comment]
>  For compatibility with older versions of CMake, option is clearing the
  normal variable 'HIGHFIVE_USE_BOOST'.

I guess the problem boils down to this. CMake is just so complicated with all these policies that just ruin compatibility instead.

Just for my understanding, if you use the old var `USE_xxx` then it works ok?

**Comment by tdegeus** [comment]
The weirdest thing happens. In order for me to give you a correct answer I retried. It now works. Apparently:

* It work with `cmake_minimum_required(VERSION 3.16)`
* It break with `cmake_minimum_required(VERSION 3.1)`

*while running with the same CMake, version 3.17*?!?

**Comment by ulfllorenz** [comment]
AFAIK This is not surprising. The minimum version tells CMake to behave like this version. If you tried to use advanced features introduced in, say, 3.6 with a minimum version set to 3.1, CMake should fail, for example.

**Comment by 1uc** [comment]
This might still be real:
https://github.com/BlueBrain/HighFive/blob/master/CMakeLists.txt#L1

only sets the policy to `3.13`. Hence, the minimum required policy might in the context above might be in the range `3.14`, ..., `3.16`.

**Comment by 1uc** [comment]
We've rewritten the HighFive CMake related code for `v3` and are close to of have merge the changes. I'm closing this issue as outdated. Please open a new Issue if the problem persists after #897 got merged.
